### PR TITLE
Adding sandbox utilities and tests

### PR DESCRIPTION
**Pull request by scalvert**
This is a first pass (maybe a last pass?) at creating the utilities necessary to create and restore a sinon sandbox for each test. 

@rwjblue @stefanpenner would appreciate the feedback.

### CODE REVIEW

File: tests/unit/sinon-sandbox-test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+test('configuring setup/restore', function(assert) {
+  assert.expect(2);
+
+  let testStartCalled = false;
+  let testDoneCalled = false;
+
+  let testEnvironment = {
+    testStart() {
+      testStartCalled = true;
```
Comment (rwjblue) [inline tests/unit/sinon-sandbox-test.js]: Can you confirm that the arguments are correct?

```
testStart(callback) {
  assert.equal(callback, createSandbox);
}
```

File: addon-test-support/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const SINON = self.sinon;
+
+export function createSandbox() {
+  QUnit.config.current.testEnvironment.sandbox = self.sinon = SINON.sandbox.create();
```
Comment (rwjblue) [inline addon-test-support/index.js]: Should be importing QUnit in this file  I think

File: tests/unit/sinon-sandbox-test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import setupSinonSandbox, { createSandbox, restoreSandbox } from 'ember-sinon-sandbox/test-support';
+
+module('Unit | ember-sinon-sandbox');
+
+test('stores sandbox created as module property', function(assert) {
+  assert.expect(3);
+
+  assert.notOk(QUnit.config.current.testEnvironment.sandbox);
+
+  createSandbox();
```
Comment (rwjblue) [inline tests/unit/sinon-sandbox-test.js]: This test needs to call restoreSandbox (or it pollutes)

Comment (scalvert) [reply tests/unit/sinon-sandbox-test.js]: Doh. It already bit me in terms of pollution...

File: tests/unit/sinon-sandbox-test.js (hunk 1 above)
Comment (scalvert) [reply tests/unit/sinon-sandbox-test.js]: Good call. Adding it.

File: addon-test-support/index.js (hunk 2 above)
Comment (scalvert) [reply addon-test-support/index.js]: Agreed.

**Comment by scalvert** [comment]
@rwjblue updated

**Pull request merged**